src/zenml/pipelines/pipeline_definition.py

```diff
@@ -806,31 +821,72 @@ def _run(
 
         with track_handler(AnalyticsEvent.RUN_PIPELINE) as analytics_handler:
             stack = Client().active_stack
-            deployment = self._create_deployment(**self._run_args)
 
-            self.log_pipeline_deployment_metadata(deployment)
-            run = create_placeholder_run(deployment=deployment)
+            # Enable or disable pipeline run logs storage
+            if self._run_args.get("schedule"):
+                # Pipeline runs scheduled to run in the future are not logged
+                # via the client.
+                logging_enabled = False
+            elif constants.handle_bool_env_var(
+                constants.ENV_ZENML_DISABLE_PIPELINE_LOGS_STORAGE, False
+            ):
+                logging_enabled = False
+            else:
+                logging_enabled = self._run_args.get(
+                    "enable_pipeline_logs", True
```

Comment:
I think it's still missing the check of `self.config.enable_pipeline_logs` as a last fallback?

Reply:
You're right, I missed that. Done.